### PR TITLE
gh-106490: Fixed `random.binomialvariate` BG algorithm inaccuracy

### PR DESCRIPTION
**Pull request by trag1c**
* Issue: gh-106490

### CODE REVIEW

**Review by sunmy2019 — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2023-07-06-20-09-15.gh-issue-106490.HC-ZvO.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed ``random.binomialvariate`` BG algorithm inaccuracy
```
Comment (sunmy2019) [inline Misc/NEWS.d/next/Library/2023-07-06-20-09-15.gh-issue-106490.HC-ZvO.rst:1]: How about adding more explanations for what part is wrong?

File: Lib/test/test_random.py [hunk 2]
```diff
@@ -1104,13 +1104,13 @@ def test_binomialvariate(self):
...
-        self.assertTrue(29_641 <= c[0] <= 33_641, c)
-        self.assertTrue(40_188 <= c[1] <= 44_188)
-        self.assertTrue(19_094 <= c[2] <= 23_094)
-        self.assertTrue(2_688  <= c[3] <= 6_688)
-        self.assertEqual(set(c), {0, 1, 2, 3, 4})
+        self.assertTrue(40_139 <= c[0] <= 44_139, c)
+        self.assertTrue(40_317 <= c[1] <= 44_317)
+        self.assertTrue(11_979 <= c[2] <= 15_979)
+        self.assertTrue(0      <= c[3] <= 3_565)
+        self.assertEqual(set(c), {0, 1, 2, 3})
```
Comment (sunmy2019) [inline Lib/test/test_random.py:1113]: This is definitely wrong. `B(n, p)` should has value `{0, 1, 2, ..., n}`

**Review by sunmy2019 — CHANGES_REQUESTED** [review]

File: Lib/test/test_random.py (hunk 2 above)
Comment (sunmy2019) [reply Lib/test/test_random.py:1113]: Binomial distrubution table for `B(4,p)`

p | .01 | .05 | .10 | .15 | .20 | .25 | .30 | .35 | .40 | .45 | .50 
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- 
 | 0 | .961 | .815 | .656 | .522 | .410 | .316 | .240 | .179 | .130 | .092 | .062 
 | 1 | .039 | .171 | .292 | .368 | .410 | .422 | .412 | .384 | .346 | .300 | .250 
 | 2 | .001 | .014 | .049 | .098 | .154 | .211 | .265 | .311 | .346 | .368 | .375 
 | 3 | .000 | .000 | .004 | .011 | .026 | .047 | .076 | .112 | .154 | .200 | .250 
 | 4 | .000 | .000 | .000 | .001 | .002 | .004 | .008 | .015 | .026 | .041 | .062 

Taken from https://www.thoughtco.com/binomial-table-n-2-through-6-3126258

**Comment by trag1c** [comment]
Closed because of https://github.com/python/cpython/issues/106490#issuecomment-1624654035

**Comment by rhettinger** [comment]
Also note that this PR fails simple statistical tests such as:

```
def test(n, p, times = 10 ** 6):
    data = [binomialvariate(n, p) for i in range(times)]
    q = 1.0 - p
    print(f'{n=}  {p=}')
    print(mean(data), n*p, 'mean')
    print(stdev(data), sqrt(n*p*q), 'stdev')
    print(mode(data), floor((n+1) * p), ceil((n+1) * p) - 1, 'mode')
```

With the suggested change, the results are wildly off the expected values:

```
n=2  p=0.4
0.400411 0.8 mean
0.48998190901427285 0.6928203230275509 stdev
0 1 1 mode

n=2  p=0.6
1.601059 1.2 mean
0.4896808331008253 0.6928203230275509 stdev
2 1 1 mode

n=5  p=0.75
4.000037 3.75 mean
0.8666797273299692 0.9682458365518543 stdev
4 4 4 mode

n=12  p=0.25
2.750983 3.0 mean
1.4347855562140959 1.5 stdev
3 3 3 mode

n=18  p=0.49
8.327508 8.82 mean
2.059205854271192 2.1208960370560366 stdev
8 9 9 mode

n=18  p=0.51
9.67141 9.18 mean
2.062238799152238 2.120896037056036 stdev
10 9 9 mode
```

**Pull request closed without merging**